src/lib/state/actions/query.js

```diff
@@ -137,37 +148,104 @@ export const resetQuery = () => {
   };
 };
 
+/**
+ * Update URL parameters.
+ * @param {object} queryState - current query state
+ * @param {object} appConfig - app config
+ * @param {boolean} shouldReplaceUrlQueryString - true if should replace the last browser history state
+ * @param {boolean} shouldUpdateUrlQueryString - true if it should add a new browser history state
+ */
+const updateURLParameters = (
+  queryState,
+  appConfig,
+  shouldReplaceUrlQueryString,
+  shouldUpdateUrlQueryString
+) => {
+  const urlHandlerApi = appConfig.urlHandlerApi;
+  if (urlHandlerApi) {
+    if (shouldReplaceUrlQueryString) {
+      urlHandlerApi.replace(queryState);
+    } else if (shouldUpdateUrlQueryString) {
+      urlHandlerApi.set(queryState);
+    }
+  }
+};
+
+/**
+ * Update query state and URL args with the new query state given by the backend response.
+ * @param {object} response - API response
+ * @param {func} dispatch - Redux `dispath` function
+ * @param {func} getState - function to get the Redux state
+ * @param {object} appConfig - app config
+ */
+const updateQueryStateAfterResponse = (
```

Comment:
I am a bit hesitant because now this file includes redux action + helper functions...maybe prefix them with `_` or we have a helpers file? WDYT?

Reply:
Honestly I would refactor this file, to create objects that encapsulate private stuff e.g. `updateQueryString obj and executeSearch obj`... but it requires more large changes. What about if I create a new PR with such refactoring?

Reply:
Yes even opening just an issue is fine with me :)